app/src/main/java/com/example/stromtracker/ui/geraete/GeraeteFragment.kt

```diff
@@ -1,125 +1,186 @@
 
-        geraeteViewModel.getAllHaushalt().observe(
+        geraeteViewModel.getAllKategorie().observe(
             viewLifecycleOwner,
-            Observer { haushalte ->
-                if (haushalte != null) {
-
-                    Log.d("TAGHaushalt", haushalte.toString())
-
+            Observer { kategorie ->
+                if (kategorie != null) {
+                    kategorieList.clear()
+                    kategorieList.addAll(kategorie)
 
+                }
+            })
 
 
-                    }
 
 
 
-            })
 
+    }
 
+    override fun onClick(v : View) {
+        when(v.id) {
+            R.id.button_geraete_add_verbraucher -> {
+                //TODO: Zwischen Haushalten unterscheiden!
+                val frag = GeraeteNewVerbraucherFragment(kategorieList, raumList)
+                val fragMan = parentFragmentManager
+                fragMan.beginTransaction().replace(R.id.nav_host_fragment, frag).addToBackStack(null).commit()
+            }
+            R.id.button_geraete_add_produzent -> {
+                val frag = GeraeteNewProduzentFragment(kategorieList, raumList)
+                val fragMan = parentFragmentManager
+                fragMan.beginTransaction().replace(R.id.nav_host_fragment, frag).addToBackStack(null).commit()
+            }
+            R.id.geraete_button_sort_verbrauch -> {
+                var sortedVerbrauch = geraeteList.sortedWith(compareByDescending {it.getJahresverbrauch()})
+                geraeteList.clear()
+                geraeteList.addAll(sortedVerbrauch)
+                viewAdapter.notifyDataSetChanged();
+                buttonSortVerbrauch.paintFlags = Paint.UNDERLINE_TEXT_FLAG
+                buttonSortName.paintFlags = 0
+                buttonSortRaum.paintFlags = 0
+
+            }
+
+            R.id.geraete_button_sort_name -> {
+                var sortedName = geraeteList.sortedWith(compareBy{it.getName().toLowerCase()})
+                geraeteList.clear()
+                geraeteList.addAll(sortedName)
+                viewAdapter.notifyDataSetChanged();
+                buttonSortName.paintFlags = Paint.UNDERLINE_TEXT_FLAG
+                buttonSortVerbrauch.paintFlags = 0
+                buttonSortRaum.paintFlags = 0
+            }
+
+            R.id.geraete_button_sort_raum -> {
+                var sortedRaum = geraeteList.sortedWith(compareBy{raumList[it.getRaumID() - 1].getName().toLowerCase()})
+                Log.d("TAGSort", sortedRaum.toString())
+                geraeteList.clear()
+                geraeteList.addAll(sortedRaum)
+                viewAdapter.notifyDataSetChanged();
+                buttonSortRaum.paintFlags = Paint.UNDERLINE_TEXT_FLAG
+                buttonSortVerbrauch.paintFlags = 0
+                buttonSortName.paintFlags = 0
+
+            }
+            else -> {
+                //Toast.makeText(v.context, String.format(Locale.GERMAN,"%d was pressed.", v.id), Toast.LENGTH_SHORT).show()
+            }
```

Comment:
Du solltest hier wenigstens ein Log ausgeben, dass irgendwas schie gelaufen ist, da das ja eigentlich nicht passieren sollte. Aber leer lassen, würde ich lassen

Reply:
Da kann ja nichts schief gehen an sich, wir wollen ja nur dass er bei bestimmten Button-Clicks was macht.

Wenn er keinen Button klickt, sondern irgendetwas anderes, dann soll doch keine Fehlermeldung o.ä kommen, da soll ja nichts passieren.

Reply:
Aber da kommt man jetzt doch gar nicht hin, außer irgendwas ist schiefgelaufen, oder verstehe ich das falsch?

Reply:
Theoretisch kommt man da nur hin wenn der OnClickListener noch auf etwas anderes als die Buttons gesetzt wurde, was ja gar nicht sein kann.

Aber an sich kann man das noch hinzufügen.

Reply:
Da gebe ich Matteo recht, das else wird eigentlich nicht benötigt und somit auch der Toast nicht unbedingt.
Das else wird, wie Matteo schon gesagt hat, nur aufgerufen wenn auf etwas ein Klick-Listener gesetzt wurde und die ID bei dem switch case in der  OnClick Funktion nicht behandelt wurde. Es ist sozusagen der default case.